kong/pdk/vault.lua

```diff
@@ -1238,23 +1240,30 @@ local function new(self)
   -- @treturn true|nil `true` after successfully rotating a secret, otherwise `nil`
   -- @treturn string|nil a string describing an error if there was one
   local function rotate_secret(old_cache_key, caching_strategy)
+    if old_cache_key:sub(1, #SECRETS_RETRY_KEY_PREFIX) == SECRETS_RETRY_KEY_PREFIX then
+      return true
+    end
```

Comment:
Hmm I'm a bit reluctant to add another shdict for it, not only because adding another shdict is a heavy option but also because the retry key itself contains secret reference info, so they should go to the `kong_secrets` shdcit

Reply:
makes sense. I was just trying  to avoid this special handling, but I get your point.